libcloud/common/aws.py

```diff
@@ -178,6 +180,97 @@ def _get_aws_auth_param(self, params, secret_key, path='/'):
         return b64_hmac.decode('utf-8')
 
 
+class V4SignedAWSConnection(AWSTokenConnection):
```

Comment:
We write a docstring and link to the AWS reference code you used to implement this for (future) reference.